app/dbCache.rb

```diff
@@ -99,6 +100,57 @@ def getViewsPerCampus
   return Hash[array.map(&:values).map(&:flatten)]
 end
 
+# Return array of chosen unit ids from contentCarousel config hash (two carousels)
+def unitsFromContentCarConfig(attrs)
+  car1 = attrs['contentCar1']
+  car2 = attrs['contentCar2']
+  units = [] 
+  for c in [car1, car2]
+    if c then units << c['unit_id'] if c['mode'] == 'unit' and c['unit_id'] and c['unit_id'] != "" end
+  end
+  return units
+end
+
+# Unit Carousel: Compile item and view counts for all configured units. Campus may configure up to two (2) units
+# {"uclalaw"=> {item_count: 3408, view_count: 30000}, "cpcc"=>...}
+def getUnitCarouselStats
+  unitsInCars = $activeCampuses.map do |id, c|
+    unit = Unit[id]
+    attrs = JSON.parse(unit.attrs)
+    x = unitsFromContentCarConfig(JSON.parse(unit.attrs))
+    next x
+  end
+  unitsInCars.reject!{ |u| u.empty? }.flatten!.uniq!
+  # This is how it would look if we were using the UnitItems table to get counts
+  # itemsInUnitCar = UnitItem.join(:items, :id=>:item_id).
+  #  where(:unit_id=>unitsInCars).exclude(:status=>'withdrawn').group_and_count(:unit_id).
+  #  map{|y| y.values}
+  # itemCounts = Hash[itemsInUnitCar.map(&:values).map(&:flatten)]
+  itemsInUnitCar = UnitCount.select_group(:unit_id).where(:unit_id=>unitsInCars).select_append{sum(:items_posted).as(item_count)}.map{|y| y.values}
+  viewsInUnitCar = UnitCount.select_group(:unit_id).where(:unit_id=>unitsInCars).select_append{sum(:hits).as(view_count)}.map{|y| y.values}
+  x = (itemsInUnitCar + viewsInUnitCar).group_by{|h| h[:unit_id]}.map{|k, v| v.reduce(Hash.new, :merge) }
+  return x.map{|h| [h[:unit_id], {:item_count=>h[:item_count], :view_count=>h[:view_count]}]}.to_h
+end
+
+def sumArrayCounts(array)
+  return array.inject(0) {|sum, h| sum + h[:count]}
```

Comment:
Someday I have to understand this inject thing. I've seen it done, I just never stopped to figure it out :)

Reply:
It should be renamed to accumulate. This is what I bookmarked a long time ago and has been very helpful: http://queirozf.com/entries/ruby-map-each-collect-inject-reject-select-quick-reference